pysbd/abbreviation_replacer.py

```diff
@@ -71,26 +71,26 @@ def replace_period_of_abbr(self, txt, abbr):
         txt = txt[1:]
         return txt
 
-
     def search_for_abbreviations_in_string(self):
         original = self.text
-        lowered = original.lower()
-        for abbr in self.lang.Abbreviation.ABBREVIATIONS:
-            stripped = abbr.strip()
-            if stripped not in lowered:
-                continue
-            abbrev_match = re.findall(
-                r"(?:^|\s|\r|\n){}".format(stripped), original, flags=re.IGNORECASE
-            )
-            if not abbrev_match:
-                continue
-            next_word_start = r"(?<={" + str(re.escape(stripped)) + "} ).{1}"
-            char_array = re.findall(next_word_start, self.text)
-            for ind, match in enumerate(abbrev_match):
-                self.text = self.scan_for_replacements(
-                    self.text, match, ind, char_array
-                )
-        return self.text
+        abbrs = "|".join([re.escape(abr.strip()) for abr in self.lang.Abbreviation.ABBREVIATIONS])
+        abbregex = re.compile(r"(?:^|\s|\r|\n)({})\b".format(abbrs), flags=re.IGNORECASE)
```

Comment:
I'm not sure if this is the right way to do the re.compile, because it will be compiled each time you call this function. Maybe it needs to be in `self.lang.Abbreviation.ABBREVIATIONS`?

Reply:
Moved ABBREVIATIONS regex to `__init__`. Need to check how moving all individual regex to re.compile first impact the performance.